packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -757,6 +727,92 @@ class IOSDevice extends Device {
     _logger.printError('');
   }
 
+  /// Find the Dart VM url using ProtocolDiscovery (logs from `idevicesyslog`)
+  /// and mDNS simultaneously, using whichever is found first. `idevicesyslog`
+  /// does not work on wireless devices, so only use mDNS for wireless devices.
+  /// Wireless devices require using the device IP as the host.
+  Future<Uri?> _discoverDartVMForCoreDevice({
+    required String packageId,
+    required bool ipv6,
+    required DebuggingOptions debuggingOptions,
+    ProtocolDiscovery? vmServiceDiscovery,
+  }) async {
+    Timer? maxWaitForCI;
+    final Completer<Uri?> cancelCompleter = Completer<Uri?>();
+
+    // When testing in CI, wait a max of 10 minutes for the Dart VM to be found.
+    // Afterwards, stop the app from running and upload DerivedData Logs to logs
+    // directory. CoreDevices are run through Xcode and launch logs are therefore
+    // found in DerivedData.
+    if (debuggingOptions.usingCISystem) {
+      maxWaitForCI = Timer(const Duration(minutes: 10), () async {
+        _logger.printError('Failed to find Dart VM after 10 minutes.');
+        await _xcodeDebug.exit();
+        final String? logsDirectoryPath = _platform.environment['FLUTTER_LOGS_DIR'];
```

Comment:
I don't think we have any prior examples for putting the environment details of LUCI into the tool. In the past I've passed in the log location if the tool really needed to know about it #78822

@christopherfujino wdyt about this?  I don't really like it in the public-facing tool but maybe I'm being arbitrary, but this repo does of course understand details about LUCI
https://github.com/flutter/flutter/blob/a1a801a48dae48cf0958c3dc6ab8df57813a51d2/dev/devicelab/lib/framework/host_agent.dart#L23-L33

Reply:
previously where we needed this we would expose a new CLI option to the tool (for example `--debug-logs-dir`) and then provide them from the devicelab runner. This also might be generally useful for adding other debugging of this kind if and only if the `--debug-logs-dir=$FLUTTER_LOGS_DIR` option is passed.

Reply:
Thanks for the feedback, I changed it to pass as a flag